integration/integrations/integration_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/integration/helpers"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/web/ui"
+)
+
+// TestIntegraitionCRUD starts a Teleport cluster and using its Proxy Web server,
+// tests the CRUD operations over the Integration resource.
+func TestIntegrationCRUD(t *testing.T) {
+	ctx := context.Background()
+
+	// Start Teleport Auth and Proxy services
+	authProcess, proxyProcess, _ := helpers.MakeTestServers(t)
+	authServer := authProcess.GetAuthServer()
+	proxyAddr, err := proxyProcess.ProxyWebAddr()
+	require.NoError(t, err)
+
+	roleWithFullAccess, err := types.NewRole("fullaccess", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			Namespaces: []string{apidefaults.Namespace},
+			Rules: []types.Rule{
+				types.NewRule(types.KindIntegration, services.RW()),
+			},
+		},
+	})
+	require.NoError(t, err)
+	require.NoError(t, authServer.UpsertRole(ctx, roleWithFullAccess))
+
+	integrationsEndpoint, err := url.JoinPath("sites", "$site", "integrations")
+	require.NoError(t, err)
+
+	// Set up User
+	username := "fullaccess"
+	user, err := types.NewUser(username)
+	require.NoError(t, err)
+
+	user.AddRole(roleWithFullAccess.GetName())
+	require.NoError(t, authServer.UpsertUser(user))
+
+	userPassword := uuid.NewString()
+	require.NoError(t, authServer.UpsertPassword(username, []byte(userPassword)))
+
+	webPack := helpers.LoginWebClient(t, proxyAddr.String(), username, userPassword)
+
+	// List integrations should return empty
+	//nolint:bodyclose // Body is closed in readAllReadCloser
+	resp, err := webPack.DoRequest(http.MethodGet, integrationsEndpoint, nil)
```

Comment:
There is probably a way to write this so that you don't have to continuously bypass the linter.

Maybe something like this:

```
func makeRequest(t *testing.T, webpack *Pack, method, endpoint string, body io.Reader) []byte {
    resp, err := webpack.DoRequest(method, endpoint, body)
    require.NoError(t, err)

    defer resp.Body.Close()

    body, err := io.ReadAll(resp.Body)
    require.NoError(t, err)

    return body
}
```

Reply:
I ended up moving the body reading into the helper function (`webPack.DoRequest`) which returns the response and the body
Let me know what you think